rustainers/src/runner/inner.rs

```diff
@@ -287,6 +328,83 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
 
+    fn get_docker_host(&self) -> Option<String> {
+        env::var("DOCKER_HOST").ok()
+    }
+
+    #[tracing::instrument(skip(self),fields(runner = %self))]
+    async fn find_host_network(&self) -> Result<Option<Network>, ContainerError> {
+        // If we're docker in docker running on a custom network, we need to inherit the
+        // network settings, so we can access the resulting container.
+        let docker_host = self.host().await?;
+        let custom_networks = self.list_networks("custom").await?;
+        let mut networks = vec![];
+        for network in custom_networks {
+            let path = ".IPAM.Config".to_string();
+            let network_configs: Vec<IpamNetworkConfig> = self.inspect(network.id, &path).await?;
+            networks.extend(
+                network_configs
+                    .into_iter()
+                    .filter(|x| {
+                        if let Some(subnet) = x.subnet {
+                            subnet.contains(IpAddr::V4(docker_host.0))
+                        } else {
+                            false
+                        }
+                    })
+                    .map(|_| Network::Custom(network.name.clone()))
+                    .collect::<Vec<Network>>(),
+            );
+        }
+        if let [network] = &networks[..] {
+            Ok(Some(network.clone()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    #[tracing::instrument(skip(self),fields(runner = %self))]
+    async fn host(&self) -> Result<Ip, ContainerError> {
+        if self.is_inside_container() {
+            self.default_gateway_ip().await
+        } else {
+            Ok(Ip(Ipv4Addr::LOCALHOST))
+        }
+    }
+
+    #[tracing::instrument(skip(self), fields(runner = %self))]
+    async fn default_gateway_ip(&self) -> Result<Ip, ContainerError> {
+        let hostname = env::var("HOSTNAME")?;
+        let host_id = hostname.parse::<ContainerId>()?;
+        let networks = self.inspect_networks(host_id).await?;
+        let mut gateways = vec![];
+        for (_, network) in networks {
+            // let network_id = name.parse::<ContainerId>()?;
+            if let Some(network_id) = network.id {
+                let containers = self.inspect_host_containers(network_id).await?;
+                // Due to short id vs long id
+                let container_ids = containers
+                    .keys()
+                    .map(std::string::ToString::to_string)
+                    .collect::<Vec<String>>();
+                if container_ids.contains(&hostname) {
+                    if let Some(gateway) = network.gateway {
+                        gateways.push(gateway);
+                    }
+                }
+            }
+        }
+        if let [gateway] = gateways[..] {
+            Ok(gateway)
+        } else {
+            Err(ContainerError::NoGateway)
+        }
+    }
+
+    fn is_inside_container(&self) -> bool {
+        Path::new("/.dockerenv").exists()
+    }
```

Comment:
Not sure this tricks works for other runner than "docker"

Reply:
I will double check what happens in the case of podman and nerdctl. But indeed my focus what to make it work with github actions, and github leverages docker.

Reply:
podman provides an equivalent with `/run/.containerenv` though it's mutually exclusive...

What exists in a docker container does not exists in a podman container and vice versa